### PR TITLE
add experimental notice for --allow-env flag in docs

### LINKED ISSUE #5034: Document the new '--allow-env' flag
https://www.elastic.co/guide/en/logstash/current/environment-variables.html needs the '--allow-env' flag documented.

### PR DESCRIPTION
**Pull request by purbon**
Fixes #5034

### CODE REVIEW

File: docs/static/configuration.asciidoc [hunk 1]
```diff
@@ -601,6 +601,9 @@ output {
 
 [[environment-variables]]
 === Using Environment Variables in Configuration
+
+This feature is _Experimental_, to enable it you will need to run logstash with the `--allow-env` flag.
```
Comment (andrewvc) [inline docs/static/configuration.asciidoc]: Experimental should be lowercase `_experimental_` here.

@purbon

**Comment by andrewvc** [comment]
LGTM pending the one case issue. Thanks @purbon!

**Comment by purbon** [comment]
@andrewvc fixed the capital letter.

**Comment by andrewvc** [comment]
Thanks @purbon

File: docs/static/configuration.asciidoc [hunk 2]
```diff
@@ -601,6 +601,9 @@ output {
 
 [[environment-variables]]
 === Using Environment Variables in Configuration
+
+This feature is _experimental_, to enable it you will need to run logstash with the `--allow-env` flag.
+
 ==== Overview
 
 * You can set environment variable references into Logstash plugins configuration using `${var}` or `$var`.
```
Comment (acchen97) [inline docs/static/configuration.asciidoc]: We also need to remove "$var"

Comment (acchen97) [inline docs/static/configuration.asciidoc]: and the first row in table below should therefore be removed too

**Comment by purbon** [comment]
@acchen97 @andrewvc fixed.

**Comment by andrewvc** [comment]
LGTM

File: docs/static/configuration.asciidoc [hunk 3]
```diff
@@ -618,16 +621,6 @@ output {
...
-[source,ruby]
-----
-input {
-  tcp {
-    port => "$TCP_PORT"
-  }
-}
-----
-
-|
```
Comment (acchen97) [inline docs/static/configuration.asciidoc]: The row has 3 columns, so believe you need to remove through 643.  Then LGTM

**Comment by purbon** [comment]
@acchen97 oh, I just merged with out you, @andrewvc told me it was ok! 

Are you @acchen97 able to clean up the docs?

**Pull request closed without merging**